### PR TITLE
OHRM5X-993 : Timesheet Action Log API

### PR DESCRIPTION
**Pull request by SandeepaOHRM**
This PR contains

- Working API endpoint to get all time-sheet action logs by time-sheet ID.
- Filters : action, date and performed user
- Unit testings and Integration testing

### CODE REVIEW

File: installer/dbscript/dbscript-2.sql [hunk 1]
```diff
@@ -4278,7 +4281,9 @@ VALUES (1, 0, 1, 0, 0, @apiv2_time_time_sheet_config_data_group_id, @admin_role_
        (1, 1, 1, 1, 1, @apiv2_time_projects_data_group_id, @admin_role_id),
        (1, 0, 0, 0, 0, @apiv2_time_project_name_validator_data_group_id, @admin_role_id),
        (1, 0, 1, 0, 1, @apiv2_time_timesheets_items_data_group_id, @admin_role_id),
-       (1, 0, 1, 0, 1, @apiv2_time_timesheets_items_data_group_id, @ess_role_id);
+       (1, 0, 1, 0, 1, @apiv2_time_timesheets_items_data_group_id, @ess_role_id),
+       (1, 0, 0, 0, 0, @apiv2_time_timesheet_action_log_data_group_id, @admin_role_id),
+       (1, 0, 0, 0, 0, @apiv2_time_timesheet_action_log_data_group_id, @ess_role_id);
```
Comment (SandeepaOHRM) [inline installer/dbscript/dbscript-2.sql:4287]: Not sure whether the role based permission is correct. @RajithaKumara please check

File: symfony/plugins/orangehrmTimePlugin/test/Dao/TimesheetDaoTest.php [hunk 2]
```diff
@@ -22,25 +22,19 @@
...
-     * @var TimesheetDao
-     */
     private TimesheetDao $timesheetDao;
-
-    /**
-     * @var string
-     */
-    protected string $fixture;
+    private int $timesheetId = 1;
+    protected string $fixtures;
```
Comment (SandeepaOHRM) [inline symfony/plugins/orangehrmTimePlugin/test/Dao/TimesheetDaoTest.php:37]: add doc comments

File: symfony/plugins/orangehrmTimePlugin/entity/Decorator/TimesheetActionLogDecorator.php [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+     */
+    public function __construct(TimesheetActionLog $timesheetActionLog)
+    {
+        $this->timesheetActionLog = $timesheetActionLog;
+    }
+
+    /**
+     * @return TimesheetActionLog
+     */
+    public function getTimesheetActionLog(): TimesheetActionLog
```
Comment (RajithaKumara) [inline symfony/plugins/orangehrmTimePlugin/entity/Decorator/TimesheetActionLogDecorator.php:47]:
```suggestion
    protected function getTimesheetActionLog(): TimesheetActionLog
```

File: symfony/plugins/orangehrmTimePlugin/entity/Decorator/TimesheetActionLogDecorator.php [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+     */
+    public function getDate(): string
+    {
+        return $this->getDateTimeHelper()->formatDateTimeToYmd($this->getTimesheetActionLog()->getDate());
+    }
+
+    /**
+     * @return string
+     */
+    public function getActionLabel(): string{
```
Comment (RajithaKumara) [inline symfony/plugins/orangehrmTimePlugin/entity/Decorator/TimesheetActionLogDecorator.php:63]: format the code

File: symfony/plugins/orangehrmTimePlugin/test/Dao/TimesheetDaoTest.php [hunk 5]
```diff
@@ -49,7 +54,7 @@ class TimesheetDaoTest extends KernelTestCase
     protected function setUp(): void
     {
         $this->timesheetDao = new TimesheetDao();
-        $this->fixture = Config::get(Config::PLUGINS_DIR) . '/orangehrmTimePlugin/test/fixtures/MyTimesheetAPITest.yml';
+        $this->fixture = Config::get(Config::PLUGINS_DIR).'/orangehrmTimePlugin/test/fixtures/MyTimesheetAPITest.yml';
```
Comment (RajithaKumara) [inline symfony/plugins/orangehrmTimePlugin/test/Dao/TimesheetDaoTest.php:57]: don't use another fixture file. create dedicated fixture file for your test to make it independent

File: installer/dbscript/dbscript-2.sql [hunk 6]
```diff
@@ -4248,7 +4248,8 @@ VALUES ('apiv2_time_time_sheet_config', 'API-v2 Time - Time Sheet Start Day Conf
        ('apiv2_time_projects', 'API-v2 Time - Projects', 1, 1, 1, 1),
        ('apiv2_time_project_name_validator', 'API-v2 Time - Project Name Validation', 1, 0, 0, 0),
        ('apiv2_time_timesheets_items', 'API-V2 Time - Timesheet items', 1, 0, 1, 0),
-       ('apiv2_time_my_timesheet', 'API-v2- Time - My Timesheet', 0, 1, 0, 0);
+       ('apiv2_time_my_timesheet', 'API-v2- Time - My Timesheet', 0, 1, 0, 0),
+       ('apiv2_time_timesheet_action_log', 'API-V2 Time - Timesheet action log', 1, 0, 0, 0);
```
Comment (RajithaKumara) [inline installer/dbscript/dbscript-2.sql:4252]:
```suggestion
       ('apiv2_time_timesheet_action_log', 'API-v2 Time - Timesheet action log', 1, 0, 0, 0);
```

File: symfony/plugins/orangehrmTimePlugin/Api/TimesheetActionLogAPI.php [hunk 7]
```diff
@@ -0,0 +1,116 @@
...
+    use TimesheetServiceTrait;
+
+    public const PARAMETER_TIMESHEET_ID = 'timesheetId';
+
+    /**
+     * @inheritDoc
+     */
+    public function getAll(): EndpointCollectionResult
+    {
+        $timesheetId = $this->getRequestParams()->getInt(
```
Comment (RajithaKumara) [inline symfony/plugins/orangehrmTimePlugin/Api/TimesheetActionLogAPI.php:48]: need to handle user permission for this

**Review by RajithaKumara — CHANGES_REQUESTED** [review]

File: symfony/plugins/orangehrmTimePlugin/test/Dao/TimesheetDaoTest.php (hunk 5 above)
Comment (SandeepaOHRM) [reply symfony/plugins/orangehrmTimePlugin/test/Dao/TimesheetDaoTest.php:57]: This one you said that we need to talk with Vageeshan as well and do the improvement

**Review by RajithaKumara — APPROVED** [review]

**Pull request merged**